src/Agent.Sdk/SecretMasking/OssSecretMasker.cs

```diff
@@ -81,11 +106,132 @@ public void Dispose()
     {
         _secretMasker?.Dispose();
         _secretMasker = null;
+        _telemetry = null;
     }
 
     public string MaskSecrets(string input)
     {
-        return _secretMasker.MaskSecrets(input);
+        _telemetry?.ProcessInput(input);
+        return _secretMasker.MaskSecrets(input, _telemetry?.DetectionAction);
+    }
+
+    public void EnableTelemetry()
```

Comment:
>EnableTelemetry [](http://example.com/codeflow?start=16&length=15)

Maybe note here this method will be call if/when `SendSecretMaskerTelemetry` knob is set appropriately?